### PR TITLE
Add mailing/distribution list support.

### PR DESCRIPTION
**Pull request by jhdielman**
Thought it might be useful to add support for admins to manage distribution lists.

### CODE REVIEW

File: contactform/ContactFormPlugin.php [hunk 1]
```diff
@@ -79,27 +79,70 @@ public function getSettingsHtml()
...
+  public function prepSettings($values) {
+
+    if (!$values)
 		{
 			$values = array();
 		}
 
 		if (is_array($values))
 		{
+      $postedSettings = craft()->request->getPost('settings');
```
Comment (takobell) [inline contactform/ContactFormPlugin.php]: These setSettings/prepSettings changes seem unrelated to the mailing list feature.  Can you explain what's going on? Why pull settings from POST here now?

File: README.md [hunk 2]
```diff
@@ -162,6 +162,41 @@ Then from your `craft/config/contactform.php` config file, you’ll need to add
...
+
+A hidden field with the given list name.
+```jinja
+<input type="hidden" name="mailingLists" value="LIST_NAME_FROM_SETTINGS">
+```
+Or, if you have more than one list you call this to add a select dropdown.
+```jinja
+{{ craft.contactForm.mailingListsSelect() }}
+```
+You can also add attributes to the select element by passing in a hash of desired attributes. You can also override the `id` attribute if need be though refrain from overriding the `name` as it will cause the mailing list to not send.
```
Comment (takobell) [inline README.md]: Just checking, but this is "hash" as in "key:value" pair... not "hash" as in the Twig `|hash` filter, correct?

File: contactform/ContactFormPlugin.php (hunk 1 above)
Comment (takobell) [inline contactform/ContactFormPlugin.php]: Maybe this is to support pulling in mailing list data via the config file?

**Comment by takobell** [comment]
The code should be formatted so that it fits the Craft 2.x code style guide. https://github.com/pixelandtonic/CodingStandards/blob/master/standards/PHP.md

**Comment by takobell** [comment]
Thanks for this... going to go ahead and close for now, but feel free to clarify/continue discussion!

File: README.md (hunk 2 above)
Comment (jhdielman) [inline README.md]: Correct, a key/value pair. I can certainly clarify :)

File: contactform/ContactFormPlugin.php (hunk 1 above)
Comment (jhdielman) [inline contactform/ContactFormPlugin.php]: This was the only way I was able to get the entered settings into the database. Each time I submitted the form with the settings in the table field the database would be empty. The order here is...
1. Take the posted value.
2. If the value is not set, take the default value.
3. Finally the config value takes precedence over all.

Hope this helps :)
